internal/config/config_file.go

```diff
@@ -51,30 +52,84 @@ func ConfigDir() string {
 func autoMigrateConfigDir(newPath string) {
 	path, err := os.UserHomeDir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 		return
 	}
 
 	path, err = homedir.Dir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 	}
 }
 
-func dirExists(path string) bool {
-	f, err := os.Stat(path)
-	return err == nil && f.IsDir()
+func StateDir() string {
+	if path := os.Getenv(XDG_STATE_HOME); path != "" {
+		path = filepath.Join(path, "gh")
+		if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+			_ = os.MkdirAll(path, 0755)
+			autoMigrateStateDir(path)
+		}
+		return path
+	}
+
+	return ConfigDir()
+}
+
+// Check default paths (os.UserHomeDir, and homedir.Dir) for existing state file (state.yml)
+// If state file exist then move it to newPath
+// TODO: Remove support for homedir.Dir location in v2
+func autoMigrateStateDir(newPath string) {
```

Comment:
Could this and `autoMigrateConfigDir` return errors, even though their call points (`ConfigDir()` and `StateDir()`) are discarding those errors? The error value would be useful for tests that directly exercise this functionality.